### PR TITLE
fix: 3817 dont run task if disabled

### LINKED ISSUE #3817: Scripts: disabled=true does not work
**What version of EMX2 are you using (see footer)**
v10.71.0

**Describe the bug**
When setting disabled=true in a scheduled script it is still executed. The only way to prevent the script from being executed is by removing the cron syntax.
 
**To Reproduce**
1. Go to 'https://emx2.dev.molgenis.org/_SYSTEM_/tasks/#/scripts'
2. Schedule the "hello world" script every 30 secs => */30 * * * * ?
3. Check jobs to see the script it running;  
4. Then put disabled=true => see that the script keeps running

**Expected behavior**
When disabled=true scheduled scripts should not be executed anymore.

### PR DESCRIPTION
**Pull request by connoratrug**
Closes #3817

What are the main changes you did:
- 'fix' task contractor
- unschedule task if if task row is changed to disabled 
- do not schedule tasks that are disabled when loading tasks from database

how to test:
- explain here what to do to test this (or point to unit tests)

todo:
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
Putting disabled=true in a scheduled "hello world" script in the preview still does not stop the script from being executed.

**Comment by connoratrug** [comment]
> Putting disabled=true in a scheduled "hello world" script in the preview still does not stop the script from being executed.

i does for me, could you describe the steps you took to test, maybe my test was wrong( note the 'play' still runs the scrip, the schedular does not )

File: backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceScheduler.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public void unschedule(String name) {
 
   public void update(ScriptTask scriptTask) {
     // if no cron we remove
-    if (scriptTask.getCronExpression() == null) {
+    if (scriptTask.getCronExpression() == null || scriptTask.isDisabled()) {
```
Comment (connoratrug) [inline backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceScheduler.java:113]: this line adds the missing check , the other code fixes the object loading from the db

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
Did a functional test:
- [x] put a scheduled script disabled=true => is not scheduled anymore
- [x] then set disabled = false => script starts running again
- [x] then disabled = true => script stops running
- [x] clear all disabled values => scripts starts running (as expected)
- [ ] disabled = true => manually run the script

Conclusion: a disabled=true script is not scheduled anymore, so that's fixed. However, it's also not possible to run it manually anymore, and I would expect that would still work, although a script is disabled.

File: backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/Task.java [hunk 2]
```diff
@@ -152,6 +153,10 @@ public T setTotal(int total) {
   }
 
   public T start() {
+    if (this.isDisabled()) {
```
Comment (mswertz) [inline backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/Task.java:156]: as discussed this morning, we don't need this I suppose if 'disabled' means 'not running via cron'

File: backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/Task.java [hunk 3]
```diff
@@ -284,6 +289,15 @@ public String getCronExpression() {
     return this.cronExpression;
   }
 
+  public T cronUserName(String cronUserName) {
+    this.cronUserName = cronUserName;
```
Comment (mswertz) [inline backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/Task.java:293]: this is not used right? it is now 'admin only'?

File: backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInDatabase.java [hunk 4]
```diff
@@ -96,40 +96,40 @@ public String submitTaskFromName(final String scriptName, final String parameter
...
-                        .parameters(parameters)
-                        .token(
-                            JWTgenerator.createTemporaryToken(
-                                systemSchema.getDatabase(),
-                                systemSchema.getDatabase().getActiveUser()))
-                        .submitUser(user)));
-          } else {
-            throw new MolgenisException("Script execution failed: " + scriptName + " not found");
+          if (scriptTask.isDisabled()) {
+            throw new MolgenisException("Script " + scriptName + " is disabled");
```
Comment (mswertz) [inline backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInDatabase.java:106]: and than this would also not be needed anymore

**Review by mswertz — APPROVED** [review]

File: backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/Task.java (hunk 3 above)
Comment (connoratrug) [reply backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/Task.java:289]: yes bit its used in code, only partially loading taks state from db into object makes it hard keep the logic working , as this issue shows.

**Review by dtroelofsprins — APPROVED** [review]
- [x]  running a disabled = true script manually

**Pull request merged**